Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -289,6 +316,88 @@ void TimeSplitter::clearAndReplace(const DateAndTime &start, const DateAndTime &
     m_roi_map.insert({start, value});
     m_roi_map.insert({stop, NO_TARGET});
   }
+  resetCache();
+}
+
+void TimeSplitter::resetCache() {
+  resetCachedPartialTimeROIs();
+  resetCachedSplittingIntervals();
+}
+
+void TimeSplitter::resetCachedPartialTimeROIs() const {
+  std::lock_guard<std::recursive_mutex> lock(m_mutex);
+  if (m_validCachedPartialTimeROIs) {
+    m_cachedPartialTimeROIs.clear();
+    m_validCachedPartialTimeROIs = false;
+  }
+}
+
+void TimeSplitter::resetCachedSplittingIntervals() const {
+  std::lock_guard<std::recursive_mutex> lock(m_mutex);
+  if (m_validCachedSplittingIntervals_All || m_validCachedSplittingIntervals_WithValidTargets) {
+    m_cachedSplittingIntervals.clear();
+    m_validCachedSplittingIntervals_All = false;
+    m_validCachedSplittingIntervals_WithValidTargets = false;
+  }
+}
+
+void TimeSplitter::rebuildCachedPartialTimeROIs() const {
+  std::lock_guard<std::recursive_mutex> lock(m_mutex);
+
+  resetCachedPartialTimeROIs();
+
+  if (empty())
+    return;
+
+  std::map<int, std::vector<DateAndTime>> partialTimeVectors;
+
+  int target{NO_TARGET};
+  DateAndTime intervalStart, intervalStop;
+  for (auto it = m_roi_map.begin(); it != std::prev(m_roi_map.end()); it++) {
```

Comment:
An open-ended `TimeSplitter` use case is not handled on the main branch.  Since you brought it up, I have added handling of that use case to `getTimeROI` and `getSplittingIntervals`.  However, it is unclear how one could build such an `m_roi_map` using existing `TimeSplitter` interface.

Reply:
I have updated the code per our discussion. The code will now throw an exception if an open-ended `TimeSplitter` is detected.